internal/runner/operation.go

```diff
@@ -267,6 +294,46 @@ func (r *Runner) pluginFactories(
 	return result, perr
 }
 
+// parse information on reattaching to unmanaged plugins out of a
+// JSON-encoded environment variable.
+func parseReattachPlugins(in string) (map[string]*goplugin.ReattachConfig, error) {
+	unmanangedPlugins := map[string]*goplugin.ReattachConfig{}
+	if in != "" {
+		in = strings.TrimRight(in, "'")
+		in = strings.TrimLeft(in, "'")
+		var m map[string]sdk.ReattachConfig
+		err := json.Unmarshal([]byte(in), &m)
+		if err != nil {
+			return unmanangedPlugins, fmt.Errorf("Invalid format for WP_REATTACH_PROVIDERS: %w", err)
+		}
+		for p, c := range m {
+			var addr net.Addr
+			switch c.Addr.Network {
+			case "unix":
+				addr, err = net.ResolveUnixAddr("unix", c.Addr.String)
+				if err != nil {
+					return unmanangedPlugins, fmt.Errorf("Invalid unix socket path %q for %q: %w", c.Addr.String, p, err)
+				}
+			case "tcp":
+				addr, err = net.ResolveTCPAddr("tcp", c.Addr.String)
+				if err != nil {
+					return unmanangedPlugins, fmt.Errorf("Invalid TCP address %q for %q: %w", c.Addr.String, p, err)
+				}
+			default:
+				return unmanangedPlugins, fmt.Errorf("Unknown address type %q for %q", c.Addr.String, p)
+			}
+			unmanangedPlugins[p] = &goplugin.ReattachConfig{
+				Protocol:        goplugin.Protocol(c.Protocol),
+				ProtocolVersion: c.ProtocolVersion,
+				Pid:             c.Pid,
+				Test:            c.Test,
+				Addr:            addr,
+			}
+		}
+	}
+	return unmanangedPlugins, nil
```

Comment:
Same here, you actually call this `parseReattachPlugins` but then use `unmanagedPlugins` as the result name, which is a smell to me that the whole thign should be one or the other (and I prefer `reattach`).